specs/emio-006.md

```diff
@@ -0,0 +1,104 @@
+# Abstract
+
+This proposes a standard to annotate curation or reputation information to stake pools
+
+# Motivation
+
+Unlike Proof of Work where regular users have no say in who gets to make blocks, Proof of Stake empowers users of the protocol to influence who is creating blocks.
+
+Users tend to choose who they delegate to based on two categories:
+
+1) **Financial incentives** (who promises the highest reward)
+2) **Moral incentives** (who is following the protocol, contributing to the community, etc.)
+
+Given that
+
+1) There are hundreds or thousands of pools to choose from
+2) *Financial incentives* are very quick to compare
+3) *Moral incentives* are very hard to compare
+
+Most users will simply choose their pool based on the *financial incentive*.
+
+This is problematic because:
+
+1) This is sometimes detrimental to the protocol (an adversary could promise high returns but attack the protocol)
+2) Empirically many users want to vote based on their morals but were not able to.
+
+Therefore we need a way to simplify choosing based on **moral incentives** without resulting in centralization.
+
+# Proposal
+
+We propose to use a standard format for encoding pool curation information. A standard format allows us to decouple software creators from curators. For example, a wallet or explorer could bundle their software with a pre-existing set of curators but also allow users to override to use their own preferred curator.
+
+### Format
+
+We propose the format follows (defined using [JSON Schema](https://json-schema.org/)).
+
+```
+{
+  "definitions": {
+    "node_flags": {
+      "$id": "#/definitions/node_flags",
+      "type": "integer",
+      "title": "Node-level Flags",
+      "description": "Flags that can be set by node software indicating automatically-detected adversarial behavior",
+      "default": 0
+     }
+  },
+  "$schema": "http://json-schema.org/schema#",
+  "type": "object",
+  "patternProperties": {
+    "^[a-f0-9]{64}$": {
+      "type": "object",
+      "properties": {
+        "node_flags": { "$ref": "#/definitions/node_flags" }
+      },
+      "additionalProperties": false
+    }
+  },
+  "additionalProperties": false
+}
+```
+
+Here is an example
+
+```json
```

Comment:
This is fine for small queries like # of pools creating forks. However, if somebody built a location service for this spec, the endpoint would return the location of every pool (in the thousands) which is not ideal.

We could specify two modes:
1) Request the full list
2) Request only values for a specific list of pools

That way clients like Yoroi can use (2) for efficient queries and explorers like Seiza can use (1). You can even paginate the request trivially on (2) if you have a lot of pools to query for but still less than the full list